src/core/xlio_extra.h

```diff
@@ -641,4 +645,163 @@ static inline struct xlio_api_t *xlio_get_api()
     return api_ptr;
 }
 
+extern "C" {
+int xlio_socket(int __domain, int __type, int __protocol);
+
+int xlio_close(int __fd);
+
+void xlio___res_iclose(res_state statp, bool free_addr);
```

Comment:
Remove this prototype. This is internal libresolv logic. Since user uses xlio_socket API explicitly, all dependent static/shared libraries will use systems sockets without offloading. So, libresolv will use kernel for socket interaction and libc __res_iclose() will close all internal sockets with kernel.

We have to overload this symbol in case of LD_PRELOAD, because XLIO may offload libresolv sockets in this scenario and kernel close() will break functionality.

Reply:
It would be shocking for SPDK or some other future client to statically link against XLIO and find out that we override the __res_iclose, and there is no way to opt out

Reply:
OK

Reply:
OK